internal/kgateway/krtcollections/policy.go

```diff
@@ -66,6 +67,10 @@ type BackendIndex struct {
 	policies  *PolicyIndex
 	refgrants *RefGrantIndex
 	krtopts   krtutil.KrtOptions
+
+	// poolReady is used to recompute the backend index when an InferencePool is ready.
+	poolReady *krt.RecomputeTrigger
```

Comment:
Perhaps consider calling `infPoolReady` to synchronize the context? 
```suggestion
	infPoolReady *krt.RecomputeTrigger
```